system-addon/lib/HighlightsFeed.jsm

```diff
@@ -26,6 +27,7 @@ ChromeUtils.defineModuleGetter(this, "PageThumbs",
 const HIGHLIGHTS_MAX_LENGTH = 9;
 const MANY_EXTRA_LENGTH = HIGHLIGHTS_MAX_LENGTH * 5 + TOP_SITES_DEFAULT_ROWS * TOP_SITES_MAX_SITES_PER_ROW;
 const SECTION_ID = "highlights";
+const SYNC_BOOKMARKS_FINISHED_EVENT = "weave:engine:sync:finish";
```

Comment:
I should have mentioned this in the bug, but listening for "weave:engine:sync:applied" (and still checking data=="bookmarks") would be slightly more efficient - that means there was almost certainly new bookmarks added by the engine (whereas sync:finish will be fired even if the engine did nothing)